src/tests/util.rs

```diff
@@ -348,21 +348,17 @@ pub fn sig_gen_multipart(
 const TRUEBOOL: CK_BBOOL = CK_TRUE;
 const FALSEBOOL: CK_BBOOL = CK_FALSE;
 
-pub fn import_object(
-    session: CK_ULONG,
-    class: CK_OBJECT_CLASS,
+pub fn make_attr_template(
     ulongs: &[(CK_ATTRIBUTE_TYPE, CK_ULONG)],
     bytes: &[(CK_ATTRIBUTE_TYPE, &[u8])],
     bools: &[(CK_ATTRIBUTE_TYPE, bool)],
-) -> KResult<CK_OBJECT_HANDLE> {
-    let mut template = Vec::<CK_ATTRIBUTE>::with_capacity(
-        1 + ulongs.len() + bytes.len() + bools.len(),
-    );
-    template.push(make_attribute!(
-        CKA_CLASS,
-        &class as *const _,
-        CK_ULONG_SIZE
-    ));
+    capacity: usize,
+) -> Vec<CK_ATTRIBUTE> {
+    let mut template = Vec::<CK_ATTRIBUTE>::with_capacity(if capacity > 0 {
+        capacity
+    } else {
+        ulongs.len() + bytes.len() + bools.len()
+    });
```

Comment:
This is just a though if it would not be better to change the `capacity` argument to something meaning "allocate N more" as in all the calls you use `ulongs.len() + bytes.len() + bools.len() + 1` or `0`. If the meaning would be "the given attributes + N", it would simplify both the calculation here to `ulongs.len() + bytes.len() + bools.len() + N` as well as for the callers. They would just provide 0 or 1.

Reply:
I was wondering if I should do that at all TBH, it is just an optimization to avoid resizing the vector in those cases where we push additional elements outside of the basic use.

But except for the import objects operations it turns out we never do.
I will just drop the whole thing.
These are just tests and optimizing a vector allocation here is kind of silly :)

Reply:
Remove and actually had to refactor how the template is built as I was taking references to objects on the stack ... which are gone after the helper is called ... oops.